pyproject.toml

```diff
@@ -67,6 +67,21 @@ features = [
   "ci",
 ]
 
+[tool.ruff]
+# smallest version ruff seems to support
+line-length = 99
+target-version = "py37"
+include = ["pyproject.toml", "lib/spack/**/*.py", "var/spack/repos/**/*.py", "bin/spack"]
+exclude = ["lib/spack/external"]
+extend-exclude = ["*.pyi"]
+
+[tool.ruff.lint]
+ignore = ["E731", "E203"]
```

Comment:
At a bare minimum I would `extend-select = ["I"]` so we can also get rid of isort, I think `UP` and `RUF` are also quite useful.